models/Category.php

```diff
@@ -154,36 +181,55 @@ public static function getMenuTypeInfo($type)
     public static function resolveMenuItem($item, $url, $theme)
     {
         $structure = [];
-        $category  = new Category();
 
         if ( ! $pageUrl = GeneralSettings::get('category_page')) {
             throw new InvalidArgumentException(
                 'Mall: Please select a category page via the backend settings.'
             );
         }
 
-        $iterator = function ($items, $baseUrl = '') use (&$iterator, &$structure, $pageUrl, $url) {
-            $branch = [];
+        if($item->type == 'mall-category') {
+            $category = self::find($item->reference);
+            if (!$category) {
+                return;
+            }
 
             $controller = new Controller();
-            foreach ($items as $item) {
-                $entryUrl               = $controller->pageUrl($pageUrl, ['slug' => $item->nestedSlug]);
-                $branchItem             = [];
-                $branchItem['url']      = $entryUrl;
-                $branchItem['isActive'] = $entryUrl === $url;
-                $branchItem['title']    = $item->name;
-
-                if ($item->children) {
-                    $branchItem['items'] = $iterator($item->children, $item->slug);
+            $entryUrl = $controller->pageUrl($pageUrl, ['slug' => $category->slug]);
+
+            $structure['url'] = $entryUrl;
+            $structure['isActive'] = $entryUrl === $url;
+            $structure['mtime'] = $category->updated_at;
+            $structure['title'] = $category->name;
+            $structure['code'] = $category->code;
```

Comment:
Can you DRY this part up by extracting a `getMenuItem($url)` method?

```

Reply:
I decided to chose the signature of this method a little bit, but requested changes should be done anyway.